### PR TITLE
Added critical logging

### PR DESCRIPTION
**Pull request by PascalDR**

### CODE REVIEW

File: pyeudiw/storage/db_engine.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import uuid
+import logging
 import importlib
 from pyeudiw.storage.base_cache import BaseCache
 from pyeudiw.storage.base_storage import BaseStorage
 
+logger = logging.getLogger("openid4vp_backend")
```
Comment (peppelinux) [inline pyeudiw/storage/db_engine.py:7]:
```suggestion
logger = logging.getLogger("openid4vp.storage.db")
```

File: pyeudiw/storage/base_storage.py [hunk 2]
```diff
@@ -2,7 +2,7 @@ class BaseStorage(object):
     def init_session(self, document_id: str, dpop_proof: dict, attestation: dict):
         NotImplementedError()
 
-    def update_request_object(self, document_id: str, request_object: dict):
+    def update_request_object(self, document_id: str, nonce: str, state: str, request_object: dict):
```
Comment (peppelinux) [inline pyeudiw/storage/base_storage.py:5]: state is optional

File: pyeudiw/storage/db_engine.py [hunk 3]
```diff
@@ -39,15 +42,24 @@ def _handle_instance(instance: dict) -> dict[BaseStorage | None, BaseCache | Non
     
     def init_session(self, dpop_proof: dict, attestation: dict):
         document_id = str(uuid.uuid4())
-        for storage in self.storages:
-            storage.init_session(dpop_proof, attestation)
+        for db_name, storage in self.storages:
+            try:
+                storage.init_session(dpop_proof, attestation)
+            except Exception as e:
+                logger.critical("Cannot write document with id {document_id} on {db_name}")
```
Comment (peppelinux) [inline pyeudiw/storage/db_engine.py:49]:
```suggestion
                logger.critical(f"Cannot write document with id {document_id} on {db_name}")
```

File: pyeudiw/storage/db_engine.py [hunk 4]
```diff
@@ -39,15 +42,24 @@ def _handle_instance(instance: dict) -> dict[BaseStorage | None, BaseCache | Non
...
             
         return document_id
 
-    def update_request_object(self, document_id: str, request_object: dict):
-        for storage in self.storages:
-            storage.update_request_object(document_id, request_object)
+    def update_request_object(self, document_id: str, nonce: str, state: str, request_object: dict):
```
Comment (peppelinux) [inline pyeudiw/storage/db_engine.py:53]: state is optional

File: pyeudiw/storage/db_engine.py [hunk 5]
```diff
@@ -39,15 +42,24 @@ def _handle_instance(instance: dict) -> dict[BaseStorage | None, BaseCache | Non
...
+        for db_name, storage in self.storages:
+            try:
+                storage.update_request_object(document_id, nonce, state, request_object)
+            except Exception as e:
+                logger.critical("Cannot update document with id {document_id} on {db_name}")
```
Comment (peppelinux) [inline pyeudiw/storage/db_engine.py:58]:
```suggestion
                logger.critical(f"Cannot update document with id {document_id} on {db_name}")
```

File: pyeudiw/storage/db_engine.py [hunk 6]
```diff
@@ -39,15 +42,24 @@ def _handle_instance(instance: dict) -> dict[BaseStorage | None, BaseCache | Non
...
 
     def update_response_object(self, nonce: str, state: str, response_object: dict):
-        for storage in self.storages:
-            storage.update_response_object(nonce, state, response_object)
\ No newline at end of file
+        for db_name, storage in self.storages:
+            try:
+                storage.update_response_object(nonce, state, response_object)
+            except Exception as e:
+                logger.critical("Cannot update document with nonce {nonce} and state {state} on {db_name}")
```
Comment (peppelinux) [inline pyeudiw/storage/db_engine.py:65]:
```suggestion
                logger.critical(f"Cannot update document with nonce {nonce} and state {state} on {db_name}")
```

File: pyeudiw/storage/mongo_storage.py [hunk 7]
```diff
@@ -63,10 +63,7 @@ def init_session(self, document_id: str, dpop_proof: dict, attestation: dict):
 
         return document_id
 
-    def update_request_object(self, document_id: str, request_object: dict):
-        nonce = request_object["nonce"]
-        state = request_object["state"]
-
+    def update_request_object(self, document_id: str, nonce: str, state: str, request_object: dict):
```
Comment (peppelinux) [inline pyeudiw/storage/mongo_storage.py:66]: state is optional

**Review by peppelinux — CHANGES_REQUESTED** [review]

**Comment by peppelinux** [comment]
coverage must be increased

we must have more than the 90%, even for these first release the threshold is set to 80%

unfortunately this PR has a coverage of 79%, so it must be increased

File: pyeudiw/storage/base_storage.py (hunk 2 above)
Comment (PascalDR) [reply pyeudiw/storage/base_storage.py]: Resolved in commit [9386806](https://github.com/italia/eudi-wallet-it-python/pull/67/commits/9386806d0f517ab0d729b7ac0d79e095f757ff53)

File: pyeudiw/storage/db_engine.py (hunk 4 above)
Comment (PascalDR) [reply pyeudiw/storage/db_engine.py]: Resolved in commit [9386806](https://github.com/italia/eudi-wallet-it-python/pull/67/commits/9386806d0f517ab0d729b7ac0d79e095f757ff53)

**Review by peppelinux — APPROVED** [review]

**Pull request merged**